### PR TITLE
Fix world saving loop

### PR DESCRIPTION
**Pull request by Skidamek**
(Try to) fix world saving loop. Needs more debugging. Implemented doggy workaround mixin. #24 
Any help is welcome

### CODE REVIEW

File: src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/MinecraftServerMixin.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+                    ChunkHolder chunkHolder = distanceManager.getChunk(pos);
+                    if (chunkHolder != null) {
+                        int loadingLevel = chunkHolder.getTicketLevel();
+                        loadingLevelMap.put(loadingLevel, loadingLevelMap.getOrDefault(loadingLevel, 0) + 1);
+                    } else {
+                        Circumnavigate.LOGGER.error("ChunkHolder is null for chunk: {}", chunkPos);
+                    }
+
+                    level.getChunkSource().chunkMap.toDrop.add(pos);
```
Comment (Skidamek) [inline src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/MinecraftServerMixin.java]: This is what vanilla code should do, but for some reason chunk loading level is too low. So this line adds all of left chunks to `toDrop` list which then vanilla code computes and actually unloads chunks.

File: src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/ChunkTrackerMixin.java [hunk 2]
```diff
@@ -23,6 +23,103 @@ protected ChunkTrackerMixin(int firstQueuedLevel, int width, int height) {
...
+//		}
+//
+//		if (level >= ChunkLevel.MAX_LEVEL) {
+//            Circumnavigate.LOGGER.warn("Level: {} ChunkPos: {} WrappedChunkPos: {}", level, chunkPos, wrappedChunkPos);
+//		}
+//
+//
+//		// make sure to not go over the max level
+//		return Math.min(ChunkLevel.MAX_LEVEL, level);
+//	}
```
Comment (Skidamek) [inline src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/ChunkTrackerMixin.java]: All of that commented code can replace mixin below, but results in the same issue anyway

**Comment by FamroFexl** [comment]
It does still appear that it is taking longer than it needs to, almost like there are still chunks it is over-processing.

**Comment by Skidamek** [comment]
Yes of course that's why it's draft and that's what I mean by doggy solution it should not be like that theres also extra delay to not change anything until we know it's looped, because if you have large enough world and little enough simulation distance and you are not near bounds you can leave world without issue since our wrapped calculations aren't even ran

**Pull request closed without merging**